rir/src/ir/Compiler.cpp

```diff
@@ -86,8 +86,13 @@ class CompilerContext {
     FunctionWriter& fun;
     Preserve& preserve;
 
+    bool profile;
+
     CompilerContext(FunctionWriter& fun, Preserve& preserve)
-        : fun(fun), preserve(preserve) {}
+        : fun(fun), preserve(preserve),
+          profile(
```

Comment:
Why would you want to read this variable on every invocation? can you make this a static member of the compiler, instead of an instance variable of the context?